Tests for Temporal issue 3141

arshaw opened the pull request:
Tests for https://github.com/tc39/proposal-temporal/issues/3141 @ptomato

ptomato commented:
Can we hit this bug with Duration.round and Duration.total as well? Something like, `Duration.from({ minutes: -59 }).total({ unit: 'years', relativeto: '2025-11-02T01:00:00-08:00[America/Vancouver]' })`

arshaw commented:
I think all is fixed now because both use DifferenceZonedDateTime under the hood, but I will add some tests to ensure.

arshaw commented:
I've added additional tests for `round` and `total` and they pass with the latest fix

arshaw commented:
Thanks for catching that @ptomato. I updated the tests to use days. I also added tests for the bugs https://github.com/tc39/proposal-temporal/issues/3148 and https://github.com/tc39/proposal-temporal/issues/3149. I also moved the tests into intl402.
